lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,49 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    method_name = get_correct_key(method_name)
+    if self.keys.include? method_name
+      self[method_name] = self[method_name].call
+    else
+      raise "Method #{method_name} is missing"
+    end
+  end
+
+  def lazy?(key)
+    key = get_correct_key(key)
+    self[key].respond_to? :call
+  end
+
+  def respond_to_missing?(method_name, include_private = false)
+    method_name = get_correct_key(method_name)
+    self.keys.include? method_name || super
+  end
+
+  private
+  def get_correct_key(hash_key)
+    correct_key = [hash_key.to_s, hash_key.to_sym].select { |key| self[key] != nil }
+    raise "Keys #{correct_key} as string and as symbol are present" if correct_key.count > 1
+    raise "Method #{hash_key} is missing" if correct_key[0].nil?
```

Comment:
You should not check it here (I mean raise an error if a key wasn't found).
For example, you call `#get_correct_key` in `#respond_to_missing?`, and it will raise an error if a key is absent, but should just return `false`.
Similar case with `#method_missing`. You have a validation about missing key there which never be run because if key is missing it will be failed earlier in `#get_correct_key`.